planet/http.py

```diff
@@ -394,26 +399,80 @@ async def _send(self, request, stream=False) -> httpx.Response:
+                    progress_bar: bool = False) -> Path:
+        """Write data to local file with limiting and retries.
 
         Parameters:
-            method: HTTP request method.
-            url: Location of the API endpoint.
+            url: Remote location url
+            filename: Custom name to assign to downloaded file.
+            directory: Base directory for file download. This directory will be
+                created if it does not already exist.
+            overwrite: Overwrite any existing files.
+            progress_bar: Show progress bar during download.
 
         Returns:
-            Context manager providing the streaming response.
+            Path to downloaded file.
+
+        Raises:
+            planet.exceptions.APIException: On API error.
+            planet.exceptions.ClientError: When retry limit is exceeded.
+
         """
-        request = self._client.build_request(method=method, url=url)
-        http_response = await self._retry(self._send, request, stream=True)
-        response = models.StreamingResponse(http_response)
+
+        async def _write():
+            async with self._client.stream('GET', url) as response:
+
+                dl_path = Path(
+                    directory,
+                    filename or _get_filename_from_response(response))
+                dl_path.parent.mkdir(exist_ok=True, parents=True)
+
+                await self._write_response(response,
+                                           dl_path,
+                                           overwrite=overwrite,
+                                           progress_bar=progress_bar)
+
+                return dl_path
+
+        async def _limited_write():
+            async with self._limiter:
+                dl_path = await _write()
+            return dl_path
+
+        return await self._retry(_limited_write)
+
+    async def _write_response(self,
+                              response,
+                              filename,
+                              overwrite,
+                              progress_bar):
+        total = int(response.headers["Content-Length"])
+
         try:
-            yield response
-        finally:
-            await response.aclose()
+            mode = 'wb' if overwrite else 'xb'
+            with open(filename, mode) as fp:
+
+                with tqdm(total=total,
+                          unit_scale=True,
+                          unit_divisor=1024 * 1024,
+                          unit='B',
+                          desc=str(filename),
+                          disable=not progress_bar) as progress:
+                    previous = response.num_bytes_downloaded
+
+                    async for chunk in response.aiter_bytes():
+                        fp.write(chunk)
+                        new = response.num_bytes_downloaded - previous
+                        progress.update(new - previous)
+                        previous = new
+                    progress.update()
```

Comment:
@jreiberkyle do you mean to put this statement within the chunk iteration? Unless you do, I don't see how progress is shown.